Main.java

```diff
@@ -109,51 +96,97 @@ static void btnCrudGoBackMethod (){
         crudRunning=false;
     }
     static void btnCrudCreateMethod(){
-
+        String result="Waiting access...";
+        try{
+            result=db1.addProduct(conn); //function passes result onto the string
+        }catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        crudOutputStream.push(result); //send result to stream for render later
     }
     static void btnCrudReadMethod(){
-
+        String result="Waiting access...";
+        try{
+            result=db1.select(conn); //function passes result onto the string
+        }catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        crudOutputStream.push(result); //send result to stream for render later
     }
     static void btnCrudUpdateMethod(){
-
+        String result="Waiting access...";
+        try{
+            result=db1.update(conn); //function passes result onto the string
+        }catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        crudOutputStream.push(result); //send result to stream for render later
     }
     static void btnCrudDeleteMethod (){
-
+        String result="Waiting access...";
+        try{
+            result=db1.delete(conn); //function passes result onto the string
+        }catch (SQLException ex) {
+            ex.printStackTrace();
+        }
+        crudOutputStream.push(result); //send result to stream for render later
     }
-    static void simRender(){

```

Comment:
I'm not sure I understand the reason for renaming simrender to apprender?

Reply:
Just renaming it to match its name with its purpose. This method now handles rendering for all frames instead of just the simulator frame